spark/src/test/scala/io/delta/tables/DeltaTableSuite.scala

```diff
@@ -126,6 +127,37 @@ class DeltaTableSuite extends QueryTest
     }
   }
 
+  private def canonicalized(df: DataFrame): LogicalPlan = df.queryExecution.analyzed.canonicalized
+
+  test("toDf - forPath") {
+    withTempDir { dir =>
+      testData.write.format("delta").save(dir.getAbsolutePath)
+      val sparkDf = spark.read.format("delta").load(dir.getAbsolutePath)
+      val tableDf = DeltaTable.forPath(spark, dir.getAbsolutePath).toDF
+      assert(canonicalized(sparkDf) === canonicalized(tableDf))
+    }
+  }
+
+  test("toDf - forName") {
+    withTempDir { dir =>
+      withTable("deltaTable") {
+        testData.write.format("delta").saveAsTable("deltaTable")
+        val sparkDf = spark.table("deltaTable")
+        val tableDf = DeltaTable.forName(spark, "deltaTable").toDF
+        assert(canonicalized(sparkDf) === canonicalized(tableDf))
+      }
+    }
+  }
+
+  test("toDf - forName - with delta.`path`") {
+    withTempDir { dir =>
+      testData.write.format("delta").save(dir.getAbsolutePath)
+      val sparkDf = spark.read.format("delta").load(dir.getAbsolutePath)
+      val tableDf = DeltaTable.forName(spark, s"delta.`$dir`").toDF
+      assert(canonicalized(sparkDf) === canonicalized(tableDf))
+    }
+  }
```

Comment:
i think adding a stacked alias test makes sense.

Reply:
Given https://github.com/delta-io/delta/pull/2094#discussion_r1336590771, what should the test check? Are you recommending a negative test, that each new alias suppresses all previous table names?

Also: Canonicalized plans don't include subquery alias... they just return their child, see [SubqueryAlias.doCanonicalize](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala#L170). But I couldn't find a better way to compare plans, that ignores expression ids that always differ.

Reply:
yeah. we dont need this any more.